components/registry-facade-api/imagespec.proto

```diff
@@ -8,16 +8,18 @@ option go_package = "github.com/gitpod-io/gitpod/registry-facade/api";
 message ImageSpec {
     // base_ref points to an image in another registry
     string base_ref = 1;
-    // ide_ref points to an image denoting the IDE to use
+    // DEPRECATED ide_ref points to an image denoting the IDE to use
     string ide_ref = 2;
     // content_layer describe the last few layers which provide the workspace's content
     repeated ContentLayer content_layer = 3;
-    // desktop_ide_ref points to an image denotign the desktop IDE to use
+    // DEPRECATED desktop_ide_ref points to an image denotign the desktop IDE to use
     string desktop_ide_ref = 4;
     // supervisor_ref points to an image denotign the supervisor to use
     string supervisor_ref = 5;
-    // desktop_ide_plugin_ref points to an image denotign the desktop IDE plugin to use
+    // DEPRECATED desktop_ide_plugin_ref points to an image denotign the desktop IDE plugin to use
     string desktop_ide_plugin_ref = 6;
+    // dynamic_layer_ref contains all layers that ide need
+    repeated string dynamic_layer_ref = 7;
```

Comment:
This name is too generic. Please name something akin to `ide_layer_ref`.
Also in subsequent code in registry-facade.